tensorflow/python/ops/array_ops.py

```diff
@@ -575,9 +579,16 @@ def _slice_helper(tensor, slice_spec, var=None):
     TypeError: If the slice indices aren't int, slice, ellipsis,
       tf.newaxis or scalar int32/int64 tensors.
   """
+  def convert_if_possible(slice_spec):
```

Comment:
Why do you need this function to convert_if_possible?

Instead test if the types are bool, np.array(dtype=np.bool), tf.Tensor(dtype=tf.bool).

I ask because a naked try...except is very dangerous.

Reply:
My motivation is, for np.array(dtype=np.bool), tf.Tensor(dtype=tf.bool) it is very easy to check. When it comes to a complicated nests of python list or tuple, such as [[True, False, False], ...], if all the elements are bool, it should be considered as a valid mask for boolean_mask. In such cases, it is not very easy to check whether they are valid or not due to a unclear amount of nests. Therefore, I decided to use try except. I am not sure if there are better way to do this, if so please let me know.

Reply:
Ok, I will keep working on it tomorrow.